src/hotspot/share/opto/memnode.cpp

```diff
@@ -1976,13 +1976,13 @@ LoadNode::load_array_final_field(const TypeKlassPtr *tkls,
   if (tkls->offset() == in_bytes(Klass::modifier_flags_offset())) {
     // The field is Klass::_modifier_flags.  Return its (constant) value.
     // (Folds up the 2nd indirection in aClassConstant.getModifiers().)
-    assert(this->Opcode() == Op_LoadI, "must load an int from _modifier_flags");
+    assert(Opcode() == Op_LoadUS, "must load an unsigned short from _modifier_flags");
     return TypeInt::make(klass->modifier_flags());
   }
   if (tkls->offset() == in_bytes(Klass::access_flags_offset())) {
     // The field is Klass::_access_flags.  Return its (constant) value.
     // (Folds up the 2nd indirection in Reflection.getClassAccessFlags(aClassConstant).)
-    assert(this->Opcode() == Op_LoadI, "must load an int from _access_flags");
+    assert(this->Opcode() == Op_LoadUS, "must load an unsigned short from _access_flags");
```

Comment:
Nit: This can be unified with line 1979 and also get rid of `this->`.

Reply:
1979 and 1985 are in different branches of an if statement (address of modifier flags vs access flags) so needs to be repeated.  But I did remove the this->